packages/millicast-sdk/src/workers/TransformWorker.worker.js

```diff
@@ -1,44 +1,78 @@
 import { addH26xSEI, extractH26xMetadata } from '../utils/Codecs'
 
-let uuid = ''
-let payload = ''
+const DROPPED_SOURCE_TIMEOUT = 2000
+const metadata = []
 let codec = ''
 let codecMap = {}
+const synchronizationSources = {}
+let synchronizationSourcesWithMetadata = []
 
-function createReceiverTransform () {
+function createReceiverTransform (mid) {
   return new TransformStream({
     start () {},
     flush () {},
     async transform (encodedFrame, controller) {
-      const frameCodec = codecMap[encodedFrame.getMetadata().payloadType] || codec.toUpperCase()
-      if (frameCodec === 'H264') {
-        const metadata = extractH26xMetadata(encodedFrame, frameCodec)
-        if (metadata.timecode || metadata.unregistered || metadata.seiPicTimingTimeCodeArray?.length > 0) {
-          self.postMessage({ metadata })
+      // eslint-disable-next-line no-undef
+      if (encodedFrame instanceof RTCEncodedVideoFrame) {
+        const frameCodec = codecMap[encodedFrame.getMetadata().payloadType] || codec.toUpperCase()
+        if (frameCodec === 'H264') {
+          const metadata = extractH26xMetadata(encodedFrame, frameCodec)
+          if (metadata.timecode || metadata.unregistered || metadata.seiPicTimingTimeCodeArray?.length > 0) {
+            self.postMessage({ mid, metadata })
+          }
         }
       }
       controller.enqueue(encodedFrame)
     }
   })
 }
 
+function clearMetadata () {
+  if (Object.keys(synchronizationSources).sort().join() === synchronizationSourcesWithMetadata.sort().join()) {
+    metadata.shift()
+    synchronizationSourcesWithMetadata = []
+  }
+}
+
+function refreshSynchronizationSources (newSyncSource) {
+  const now = new Date().getTime()
+  synchronizationSources[newSyncSource] = now
+
+  const sourcesToDelete = Object.keys(synchronizationSources)
+    .filter(source => (now - synchronizationSources[source] > DROPPED_SOURCE_TIMEOUT))
+
+  sourcesToDelete.forEach(source => {
+    delete synchronizationSources[source]
+    delete synchronizationSourcesWithMetadata[source]
+  })
+  clearMetadata()
+}
+
 function createSenderTransform () {
   return new TransformStream({
     start () {},
     flush () {},
     async transform (encodedFrame, controller) {
-      if (uuid && payload) {
-        try {
-          // Add h265 regex when ready
-          if (!/(h26[4])/.test(codec)) {
-            throw new Error('Sending metadata is not supported with any other codec other than H.264')
+      // eslint-disable-next-line no-undef
+      if (encodedFrame instanceof RTCEncodedVideoFrame) {
+        const frameMetadata = encodedFrame.getMetadata()
+        const newSyncSource = frameMetadata.synchronizationSource
+
+        refreshSynchronizationSources(newSyncSource)
+
+        if (!synchronizationSourcesWithMetadata.includes(newSyncSource) && metadata.length) {
+          try {
+            // Add h265 regex when ready
+            if (!/(h26[4])/.test(codec)) {
```

Comment:
how about H264 here (uppercase)?

Reply:
This codec comes from the `options.codec` of the connect Publish method, that is set to be lower case